components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/services/SessionManagementService.java

```diff
@@ -35,19 +35,15 @@
 
     public boolean removeSession(String sessionId) {
 
-        if (StringUtils.isBlank(sessionId)) {
-            return false;
-        }
-        // Retrieve session information from cache in order to publish event
-        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(sessionId);
-        terminateSession(sessionContext, sessionId);
-        return true;
+        ServerSessionManagementService serverSessionManagementService =
+                FrameworkServiceDataHolder.getInstance().getServerSessionManagementService();
+        return serverSessionManagementService.removeSession(sessionId);
     }
 
     /**
      *
      * Terminates the requested session, after validating whether the session belongs to the logged in user.
-     *
+     * Use UserSessionManagementService to terminate the session of the current user
```

Comment:
Add deprecated java doc annotation and link to that class and method

Reply:
Added in 26c25fc